students/V_Medina/lesson_04/mailroom_2.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+"""
+Victor Medina
+Date: 1/12/2019
+Assignment 4: Mailroom Part 2
+"""
+import sys
+
+
+def send_thank_you(donors):
+    """
+
+    :param donors: dictionary
+        Dictionary of donor names with a value of a list of donation amoounts
+    :return: donors: dictionary
+        updated dictionary with new donation amount and/or new donor
+    """
+    print(donors)
+    name = input("What's the name? ")
+    if name == 'list':
+        for donor in donors.keys():
+            print(donor)
+        name = input("What's the name? ")
+    if any(name in donor_info for donor_info in donors):
+        donor = name
+    else:
+        donors[name] = []
+        donor = name
+
+    donation_amount = int(input('Donation Amount? '))
+    donors.get(donor).append(donation_amount)
+    print(donors)
+    print('Thank you {} for your generous donation of {} dollars!'.format(donor, donation_amount))
+    return donors
+
+
+def create_report(donors):
+    """
+    function that prints out a nicely formatted report summarizing donor list, total donations, average donation and
+    number of donations
+    :param donors: dictionary
+        Dictionary of donor names with a value of a list of donation amoounts
+    :return: None
+        function prints out report. No return is needed
+    """
+    donor_report={}
+    for donor in donors:
+        donor_report[donor] = dict(total_donation=sum(donors[donor]), num_donations=len(donors[donor]),
+                                   avg_donation=sum(donors[donor]) / len(donors[donor]))
+    print(donor_report)
+    print('Donor Name      | Total Given |   Num Gifts  | Average Gift')
+    print('------------------------------------------------------------')
+    for donor in donor_report:
+        print('{:<15} ${:>13} {:>14} ${:>13,.2f}'.format(donor, donor_report[donor]['total_donation'],
+                                                         donor_report[donor]['num_donations'],
+                                                         donor_report[donor]['avg_donation']))
+    return None

```

Comment:
* no need to `return None` 
* I don't see sorting feature here? please add the missing feature.